packages/mdc-textfield/foundation.ts

```diff
@@ -22,127 +22,110 @@
-  get shouldAlwaysFloat_() {
+  private get shouldAlwaysFloat_(): boolean {
     const type = this.getNativeInput_().type;
     return ALWAYS_FLOAT_TYPES.indexOf(type) >= 0;
   }
 
-  /** @return {boolean} */
-  get shouldFloat() {
-    return this.shouldAlwaysFloat_ || this.isFocused_ || !!this.getValue() || this.isBadInput_();
+  get shouldFloat(): boolean {
+    return this.shouldAlwaysFloat_ || this.isFocused_ || Boolean(this.getValue()) || this.isBadInput_();
   }
 
   /**
-   * {@see MDCTextFieldAdapter} for typing information on parameters and return
-   * types.
-   * @return {!MDCTextFieldAdapter}
+   * See {@link MDCTextFieldAdapter} for typing information on parameters and return types.
    */
-  static get defaultAdapter() {
-    return /** @type {!MDCTextFieldAdapter} */ ({
-      addClass: () => {},
-      removeClass: () => {},
-      hasClass: () => {},
-      registerTextFieldInteractionHandler: () => {},
-      deregisterTextFieldInteractionHandler: () => {},
-      registerInputInteractionHandler: () => {},
-      deregisterInputInteractionHandler: () => {},
-      registerValidationAttributeChangeHandler: () => {},
-      deregisterValidationAttributeChangeHandler: () => {},
-      getNativeInput: () => {},
-      isFocused: () => {},
-      activateLineRipple: () => {},
-      deactivateLineRipple: () => {},
-      setLineRippleTransformOrigin: () => {},
-      shakeLabel: () => {},
-      floatLabel: () => {},
-      hasLabel: () => {},
-      getLabelWidth: () => {},
-      hasOutline: () => {},
-      notchOutline: () => {},
-      closeOutline: () => {},
-    });
-  }
+  static get defaultAdapter(): MDCTextFieldAdapter {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      addClass: () => undefined,
+      removeClass: () => undefined,
+      hasClass: () => true,
+      registerTextFieldInteractionHandler: () => undefined,
+      deregisterTextFieldInteractionHandler: () => undefined,
+      registerInputInteractionHandler: () => undefined,
+      deregisterInputInteractionHandler: () => undefined,
+      registerValidationAttributeChangeHandler: () => new MutationObserver(() => undefined),
+      deregisterValidationAttributeChangeHandler: () => undefined,
+      getNativeInput: () => null,
+      isFocused: () => false,
+      activateLineRipple: () => undefined,
+      deactivateLineRipple: () => undefined,
+      setLineRippleTransformOrigin: () => undefined,
+      shakeLabel: () => undefined,
+      floatLabel: () => undefined,
+      hasLabel: () => false,
+      getLabelWidth: () => 0,
+      hasOutline: () => false,
+      notchOutline: () => undefined,
+      closeOutline: () => undefined,
+    };
+    // tslint:enable:object-literal-sort-keys
+  }
+
+  private isFocused_ = false;
+  private receivedUserInput_ = false;
+  private isValid_ = true;
+  private useNativeValidation_ = true;
+
+  private readonly inputFocusHandler_: () => void;
+  private readonly inputBlurHandler_: EventListener;
```

Comment:
All these EventListeners should be mdc-base/SpecificEventListeners<>

Reply:
Done